crates/proof-of-sql/src/base/proof/transcript.rs

```diff
@@ -0,0 +1,27 @@
+use crate::base::scalar::Scalar;
+use zerocopy::{AsBytes, FromBytes};
+
+/// A public-coin transcript.
+///
+/// This trait contains several method for adding prover messages and computing verifier challenges.
+///
+/// Implementation note: this is intended to be implemented via [super::transcript_core::TranscriptCore] rather than directly.
+#[allow(dead_code)]
+pub trait Transcript {
+    /// Creates a new transcript
+    fn new() -> Self;
+    /// Appends the provided messages by appending the reversed raw bytes (i.e. assuming the message is bigendian)
+    fn extend_as_be<M: FromBytes + AsBytes>(&mut self, messages: impl IntoIterator<Item = M>);
+    /// Appends the provided messages by appending the raw bytes (i.e. assuming the message is littleendian)
+    fn extend_as_le<'a, M: AsBytes + 'a>(&mut self, messages: impl IntoIterator<Item = &'a M>);
```

Comment:
Hmm asymmetry between APIs for be and le do not look intuitive.

Reply:
Yeah, I know. I think I need to rename them. Suggestions?

The reason for the asymmetry is because `le` can be by reference because it is just looking at the raw bytes.
`be` needs to either copy or own the data since it has to reverse the bytes. I prefer forcing the caller to copy, so it needs to own.

Additionally, these are the signatures that I am using downstream.

Reply:
What about `reverse_and_extend_as_be` / `reverse_and_extend_scalars_as_be`?